File: kirun-js/src/engine/runtime/expression/operators/binary/ArithmeticSubtractionOperator.ts
import { Operation } from '../../Operation';
import { BinaryOperator } from './BinaryOperator';

export class ArithmeticSubtractionOperator extends BinaryOperator {
    public apply(t: any, u: any): any {
        this.nullCheck(t, u, Operation.SUBTRACTION);
        return t + u;
    }
}

File: kirun-py/src/kirun_py/runtime/expression/operators/binary/arithmetic_subtraction_operator.py
from __future__ import annotations

from typing import Any

from kirun_py.runtime.expression.operators.binary.binary_operator import BinaryOperator


class ArithmeticSubtractionOperator(BinaryOperator):

    def apply(self, t: Any, u: Any) -> Any:
        from kirun_py.runtime.expression.operation import Operation
        self.null_check(t, u, Operation.SUBTRACTION)
        return t - u
